docs/docs/guide/tutorial.md

```diff
@@ -0,0 +1,285 @@
+---
+order: 2.1
+title: React Tutorial
+next: ./platform
+---
+
+# React tutorial
+
+This tutorial walks you through creating a simple shared counter application in `react` using DXOS to share counter state between peers.
+
+The code completed application can be found [here](https://github.com/dxos/shared-counter) and you can play with the app [here](https://shared-counter.netlify.app/).
+
+::: note In this tutorial, we will:
+
+- Build a `react` app using a [DXOS app template](#create-an-app).
+- Use [ECHO](#updating-the-counter) for real-time state consensus.
+- Create a decentralized identity with [HALO](#creating-a-user-identity).
+- [Deploy](#deploying-the-app) the app to Netlify.
+
+:::
+
+## Creating a DXOS app
+
+DXOS works in any Node.js or Browser environment. There is a [TypeScript API](typescript) and a [`react` API](react).
+
+Ensure `node -v` is at version 18 or higher (we recommend [Node Version Manager](https://github.com/nvm-sh/nvm)).
+
+First, create a new empty folder:
+
+```bash
+mkdir shared-counter
+cd shared-counter
+```
+
+We have a few [app templates](./cli/app-templates.md) that are designed to get you going quickly. They are based on [`vite`](https://vitejs.dev/), [`typescript`](https://www.typescriptlang.org/), [`react`](https://reactjs.org/), [`tailwind`](https://tailwindcss.com/), [`pwa`](https://vite-pwa-org.netlify.app/), and other opinions.
+
+For this guide, we're going to start with the [`bare`](./cli/app-templates.md#bare-template) template and create a simple shared counter. Initialize the app with `npm create`:
+
+```bash
+npm create @dxos/bare
+```
+
+::: note
+If you encounter an error with `EINVALIDPACKAGENAME` it's likely the npm/node versions are out of date. Ensure `node -v` is 18 or higher and `npm -v` is 9 or higher.
+:::
+
+Then, use your favorite package manager such as `yarn`, `npm` or `pnpm`:
+
+```bash
+npm install
+npm run serve
+```
+
+This will start the development server and print its URL to the console.
+
+The bare template is an empty `react` app wrapped in some DXOS goodness. Open the `App.tsx` file and you'll see this:
+
+```tsx
+const config = async () => new Config(await Dynamics(), Local(), Defaults());
```

Comment:
API comment: is this required?

Reply:
I believe so. It gets passed into `ClientProvider`. Are you suggesting there should be a default?

Reply:
we could push these three things into the `new Config()` constructor and make these three the default only when the call is parameterless `new Config()`